torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -393,12 +393,30 @@ class ScriptModuleSerializer {
     // Then we serialize all code info.
     writeCode(module.type());
     // The tensor constants from the code are written to a separate archive
-    // so loading the code does not depend on loading the data
+    // (constant archive) so loading the code does not depend on loading the
+    // data
     std::vector<IValue> ivalue_constants(
         constant_table_.begin(), constant_table_.end());
+    TensorIndexMap constants_from_jit;
```

Comment:
Given that serialize is a member fo ScriptModuleSerializer shouldn't we just directly write `tensors_archive_table_` instead of first writing to constants_from_jit?

Reply:
And if you are making an intermediate copy so that only during writeByteCode you can set tensors_archive_table_, which is used in writeArchive that I would suggest adding a flag like `use_tensor_archive_table_` which is set by writeByteCode.

Reply:
Ah very good point! Thanks for the detailed review :D